staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness.go

```diff
@@ -163,6 +166,8 @@ func WithPriorityAndFairness(
 		}
 
 		// find the estimated "width" of the request
+		// TODO: Maybe just make it costEstimator and let it return additionalLatency too for the watch?
```

Comment:
The costs are two-dimensional, so should not be summarized by the word "width" (which refers to only one of the two dimensions).

Reply:
Is your comment about the naming only (i.e. we should rename `width` to something else (e.g. `cost`)) or something else. I guess I'm not fully following.

Reply:
[If it's just about naming - can we merge that since code freeze is coming in a day (and I will open a follow up for the naming change.]

Reply:
My objection is primarily about naming.
Regarding grouping, there are additional questions that could be asked.

Reply:
There are two places where the struct currently named `Width` appear:
- in the struct named `RequestDigest`
- along with `hashValue uint64, flowDistinguisher, fsName string, descr1, descr2 interface{}, queueNoteFn fq.QueueNoteFn` in many places in the queueset code, which arguably should be collected into a struct.

If indeed that longer parameter list is coalesced into a struct, then the struct currently named `Width` will always appear inside another struct and could arguably be replaced there by its current one field (future two or three fields).  The potential third field I am thinking about is the estimated execution duration.  Note that today this is a constant in queuest but a variable (derived from deadline in context) in the higher layer.  Not entirely sure right now which is best to do in queuest.

Reply:
It's also part of WidthEstimator: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/width.go

which I think should return a single thing. How about calling it Cost (it would be request.Cost) [and renaming to CostEstimator]?
Do you have a better suggestions?


Given the above I think we still need that. I agree that we may want to collect the params into single struct, but I don't have a reasonable proposal how to name that struct.

Reply:
We discussed this yesterday with Mike and the conclusion was that the above arguments make sense.

So we just want to rename it (WorkEstimate [or less preferrably Work if it won't appear to really be estimate across the code] were decided).

I will work on renaming in the upcoming days.

Reply:
I opened https://github.com/kubernetes/kubernetes/pull/103664 to address this.